File: tests/fixtures/typescript/src/gamma.ts
/**
 * Case c (D3): project-wide name collision.
 *
 * This is script-mode too (no import/export), sharing the same global scope
 * as alpha.ts/beta.ts at runtime — plain scripts have no file scoping at
 * all, which is the real-world hazard ES modules exist to prevent, and the
 * most realistic way an unqualified TS/JS collision like D3 arises in code
 * that actually runs. (An ES-module `import { helper }` from two
 * differently-named files would need an alias and simply couldn't collide
 * this way — see ../README.md "Contract interpretations".) codegraph's
 * resolver, which does no scope/usage analysis, must surface this as
 * AMBIGUOUS rather than blending or guessing (D3's exact failure mode).
 */
function dispatch(): void {
  helper();
}

File: tests/fixtures/go/gamma.go
package main

// Case c (D3): project-wide name collision.
//
// Dot-imports bring both alpha.Helper and beta.Helper into this file's
// unqualified scope — the only Go import form that makes a bare,
// syntactically-ambiguous cross-package call possible at all (ordinary Go
// calls are always package-qualified and therefore never ambiguous; see
// ../README.md "Contract interpretations"). `go build` would reject this
// outright ("Helper redeclared in this block"), but it is syntactically
// valid Go, and codegraph's resolver — which does no compile-time
// redeclaration checking — must independently report AMBIGUOUS rather than
// blending (D3's exact failure mode) or silently picking one.
import (
	. "fixture-go/alpha"
	. "fixture-go/beta"
)

func Dispatch() {
	Helper()
}
